### ISSUE TITLE
beta 3.9389 is crashing

### ISSUE THREAD

**Issue by junrt** [domoticz/domoticz#2369]
When the database is reloaded domoticz is chrashing

**Comment by emontnemery** [comment]
@junrt, please provide stack trace, for example:
```
gdb ./domoticz
run
<wait for segmentation fault>
bt
```

**Comment by heggink** [comment]
Noticed 2 things with 9389: 1) it also crashes for me and 2) domoticz process memory usage doubled compared to 9300. 
![domo process mem](https://user-images.githubusercontent.com/3987735/39565659-15942e04-4eb9-11e8-890a-af5b9e256f6f.JPG)

Running valgrind on my ubuntu test box as we speak. Let's see.

**Comment by Tontze** [comment]
same behaviour here too

**Comment by emontnemery** [comment]
@junrt , @heggink , @Tontze How do you trigger a database reload?

**Comment by heggink** [comment]
@emontnemery no idea. I had one crash 8ish hrs after upgrade on my pi3 and couldn't get my Ubuntu box to crash under valgrind so no clue what caused the crash and how to reproduce.

**Comment by junrt** [comment]
After a fresh install rasbian , than the curl domoticz bash, cd domoticz ./updatebeta, 
than i do the database release throw the webpage settings.

I think the crashing started  after a raspi-config update.

Because when i start fresh without te beta en db reload it crashed also

**Comment by junrt** [comment]
New Setup:
Clean install Rasbian last stable version
Domoticz last beta throw ./updatebeta
No db restore
No plugins for / sonos / sony
All Hardware is set / rfxcom / zwave usb / netatmo / toon / hardware monitor

And now we will wait

**Comment by xztraz** [comment]
got this

Version: 3.9390
Build Hash: 8d290f21
Compile Date: 2018-05-06 12:50:59
dzVents Version: 2.4.5
Python Version: 3.4.2 (default, Oct 19 2014, 14:03:53) [GCC 4.9.1]


maybe not same bug but a crash.
```

2018-05-07 01:01:25.019  (RfLink) Rain (Ute-Regn)
2018-05-07 01:01:25.030  HttpLink: sending global variable outside_rainrate with value: 0.0
[Thread 0x660ff440 (LWP 16484) exited]
[New Thread 0x654ff440 (LWP 16501)]
[New Thread 0x67c23440 (LWP 16502)]
[Thread 0x654ff440 (LWP 16501) exited]
[New Thread 0x660ff440 (LWP 16503)]
[New Thread 0x654ff440 (LWP 16504)]

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x758eb440 (LWP 2425)]
strlen () at ../ports/sysdeps/arm/armv6/strlen.S:26
26      ../ports/sysdeps/arm/armv6/strlen.S: No such file or directory.
(gdb) bt
#0  strlen () at ../ports/sysdeps/arm/armv6/strlen.S:26
#1  0x7618b274 in PyUnicode_FromFormatV () from /usr/lib/arm-linux-gnueabihf/libpython3.4m.so
#2  0x76281fbc in PyErr_Format () from /usr/lib/arm-linux-gnueabihf/libpython3.4m.so
#3  0x76191884 in PyObject_GetAttr () from /usr/lib/arm-linux-gnueabihf/libpython3.4m.so
#4  0x762eee74 in PyEval_EvalFrameEx () from /usr/lib/arm-linux-gnueabihf/libpython3.4m.so
#5  0x762f4264 in PyEval_EvalFrameEx () from /usr/lib/arm-linux-gnueabihf/libpython3.4m.so
#6  0x762f570c in PyEval_EvalCodeEx () from /usr/lib/arm-linux-gnueabihf/libpython3.4m.so

```

**Comment by emontnemery** [comment]
@xztraz Can you please post a bit more of the log, let's say the last 50 lines or so?

**Comment by heggink** [comment]
latest beta (9415) has a signal 6 again (sigh)...

**Comment by xztraz** [comment]
@emontnemery sure. from backtrace you mean or domoticz.log?

**Comment by emontnemery** [comment]
@xztraz Both more of the backtrace (it should be longer than 6 entries) and more of the domoticz log :)

**Comment by xztraz** [comment]
i only got 6 entries from bt. can i force more? i think it said repeating something..

**Comment by emontnemery** [comment]
@xztraz Never mind then, but do you have a bit more of the domoticz log?
Your crash could potentially be fixed by #2377.

**Comment by xztraz** [comment]
i'll get more log next crash. i'm on latest rpi build all the time and its working a lot better now.

**Comment by heggink** [comment]
On 9433 which seems really stable to me so far. No crashes so nothing to debug ATM.

**Comment by heggink** [comment]
back to debugging: latest beta 9517 throws both signal 6 and signal 11's again :-(

**Comment by gizmocuz** [comment]
@heggink , Has this been resolved?

**Comment by heggink** [comment]
for me, yes. You fixed thru #2440

**Comment by gizmocuz** [comment]
Great! Thanks for the feedback!

**Comment by heggink** [comment]
Bloody hell. You have some interesting timing: https://www.domoticz.com/forum/viewtopic.php?f=6&t=19022
Just got in straight after you closed this.

**Comment by gizmocuz** [comment]
@heggink , that user does not even report a version, so it might be with an old beta

**Comment by gizmocuz** [comment]
or he/she is using an incompatible python version, you need python 3.5.3

**Comment by heggink** [comment]
he says latest beta: "i am running the latest beta 9589" hence my surprise.
Re py 3.5.3, I am also running 3.4 due to he sig11 issues with 3.5.3. Are you saying that these have been resolved and we should upgrade again to 3.5.3?

**Comment by gizmocuz** [comment]
there was a issue here, that says it was resolved when using 3.5.3
The arm build systems also use 3.5.3
That might solve it... i don't use python so i can not tell

**Comment by gizmocuz** [comment]
I see he is using python 3.5.3, does he need 3.4 perhaps ?
